text/0083-rawkv-change-data-capture.md

```diff
@@ -0,0 +1,152 @@
+
+**Resolved-ts** is generated by every leader of subscribed regions of TiKV in every `1` second by default. To calculate the **resolved-ts**, leaders will trace all **RawKV** entries "inflight", *i.e.* the entries have been in the write procedure (mainly includes scheduling, Raft propose/commit/apply, *etc.*), but not yet been observed. If there are some "inflight" entries, **resolved-ts** should be smaller than the minimum timestamp of them. Otherwise, **resolved-ts** equals to current TSO. On **TiKV-CDC** side, **resolved-ts** of a replication task is the minimum **resolved-ts** of all the regions it subscribes.
+
+**4th**, for easier implementation and better performance, **RawKV CDC** will only provide [Eventual Consistency] and [At-Least-Once] delivery.
+
+### Availability and Scalability
+
+**TiKV-CDC** is designed to be an architecture with single **Owner** and multiple workers (named as **Capture**).
+
+<img src="../media/rawkv-cdc-arch-simple.png" alt="TiKV-CDC Architecture Overview" width="60%"/>
+
+**Owner** is in charge of user interface, **Captures** management, and replication tasks dispatching. **Captures** are responsible for the data processing of replication tasks.
+
+**Owner** is elected using [ETCD]. **Owner** and **Captures** are stateless, all metadata (mainly including tasks parameters and states) is stored in [ETCD]. So the availability and scalability are archived depending on [ETCD]. As size of metadata is small and update frequency is low, [ETCD] should be able to handle the workload.
+
+For availability, we utilize a **Checkpoint** mechanism. **Checkpoint** is the latest **resolved-ts** that has been written to downstream successfully, and persist in [ETCD]. When **TiKV-CDC** cluster starts after planned or unexpected stop, replication tasks will resume from the **Checkpoint**.
+
+**Checkpoint** will be affected by **Garbage Collection**. **Checkpoint** requires that all versions since **Checkpoint** are kept, otherwise **TiKV-CDC** would miss some changes. So **TiKV-CDC** will set a **Service Safe Point** on timestamp the same as **Checkpoint**, with **TTL** defaults to `24` hours. The **TTL** means that **TiKV-CDC** and replicate tasks should be resumed in no more than **TTL**, otherwise the **Checkpoint** will be invalid.
+
+Finally, in this proposal, one replication task will be processed in only one **Capture**, to make it right more easily. Splitting a task to sub-tasks will get better performance and scalability, but we must be caution to avoid any overlap or hole among sub-tasks, otherwise the correctness will be violated. Sub-tasks will be considered later.
```

Comment:
```suggestion
Finally, in this proposal, one replication task will be processed in only one **Capture**, to make it right more easily. Splitting a task to sub-tasks will get better performance and scalability, but we must be cautious to avoid any overlap or hole among sub-tasks, otherwise the correctness will be violated. Sub-tasks will be considered later.
```